addons/mail_bot/models/res_users.py

```diff
@@ -25,12 +26,17 @@ class ResUsers(models.Model):
     def SELF_READABLE_FIELDS(self):
         return super().SELF_READABLE_FIELDS + ['odoobot_state']
 
-    def _init_messaging(self, store):
+    @api.model
+    def _on_webclient_bootstrap(self):
+        if self.odoobot_state in [False, 'not_initialized'] and self.env.user and self.env.user._is_internal():
+            self.env.user._init_odoobot()
+
+    @api.model
+    def _init_store_data(self, store: Store, /):
         odoobot_onboarding = False
-        if self.odoobot_state in [False, 'not_initialized'] and self._is_internal():
+        if self.odoobot_state in [False, 'not_initialized'] and self.env.user and self.env.user._is_internal():
             odoobot_onboarding = True
-            self._init_odoobot()
-        super()._init_messaging(store)
+        super()._init_store_data(store)
```

Comment:
A variable for `odoobot_onboarding` is probably not necessary anymore, the value can be directly assigned.